rpc.go

```diff
@@ -247,6 +197,35 @@ func (r *rpc) Clear(in *kvv1.Request, _ *kvv1.Response) error {
 	return nil
 }
 
+func getStorage(in *kvv1.Request, span trace.Span, r *rpc, op errors.Op) (kv.Storage, error) {
+	if in.GetStorage() == "" {
+		e := errors.Str("no storage provided")
+		span.RecordError(e)
+		return nil, errors.E(op, e)
+	}
+
+	storage, ok := r.storages[in.GetStorage()]
+
+	if !ok {
+		e := errors.Errorf("no such storage: %s", in.GetStorage())
+		span.RecordError(e)
+		return nil, errors.E(op, e)
+	}
+
+	return storage, nil
+}
+
+func composeKeys(in *kvv1.Request) []string {
+	ln := len(in.GetItems())
+	keys := make([]string, 0, ln)
```

Comment:
if you know len, you may access slice by index instead of appending.

Reply:
YES!

Reply:
Len is a second arg. Third is a cap. In this approach, you'll have an NPE panic.

Reply:
![blindssuck](https://github.com/user-attachments/assets/06cdabe6-0f01-4468-9d28-d117b985e533)

Reply:
I made the stupid bug but tests are green. WTF?

Reply:
Tests were not green: [link](https://github.com/roadrunner-server/kv/actions/runs/13962553153/job/39086428372#step:6:148)